compiler/rustc_codegen_ssa/src/back/symbol_export.rs

```diff
@@ -124,17 +124,26 @@ fn reachable_non_generics_provider(tcx: TyCtxt<'_>, cnum: CrateNum) -> DefIdMap<
             } else {
                 symbol_export_level(tcx, def_id.to_def_id())
             };
+            let codegen_attrs = tcx.codegen_fn_attrs(def_id.to_def_id());
             debug!(
                 "EXPORTED SYMBOL (local): {} ({:?})",
                 tcx.symbol_name(Instance::mono(tcx, def_id.to_def_id())),
                 export_level
             );
-            (def_id.to_def_id(), export_level)
+            (def_id.to_def_id(), SymbolExportInfo {
+                level: export_level,
+                tls: codegen_attrs.flags.contains(CodegenFnAttrFlags::THREAD_LOCAL),
```

Comment:
Is there any reason why `#[thread_local]` would be combined with `#[used]`?

Reply:
No. But we still would like to support export of a `#[no_mangle] #[thread_local]` static.